R/combineSNV_CNA.R

```diff
@@ -0,0 +1,157 @@
+# combineSNV_CNA.R
+
+#' Combine mutation data from CNA and SNV
+#'
+#' \code{combineSNV_CNA} loads the vector of preprocessed dataset from filtered CNA or SNV files,
+#'  creates hash keys corresponding to the input file type, combine the corresponding mutation count
+#'  to each gene and save to a gX format file as output file.
+#'
+#' Details.
+#' @section gX files:
+#'  gX rds files contain data of SNV and CNA combined for each gene as one object, sorted per
+#'  gene.
+#'
+#'
+#' @param fNameCNA A vector of path of fully qualified file Names of SNV rds files.
+#' @param fNameSNV A vector of path of fully qualifed filenames of CNA rds files.
+#' @param fgX The fully qualified filename/path of a gX output file, by default is gX.rds.
+#' @param silent Boolean option for writing combining process information to console, FALSE by default.
+#' @param writeLog Boolean option for log results, TRUE by default.
+#'
+#'
+#' @examples
+#' \dontrun{combine(fNameSNV="SNV.rds", fNameCNA="CNA.rds", fgX="combined_data.rds", silent=TRUE, writeLog=FALSE)}
+#'
+#' @export
+combineSNV_CNA <- function(fNameSNV=c(), fNameCNA=c(), fgX="gX.rds", silent=FALSE, writeLog=TRUE) {
+    # parameter validation
+    if (length(fNameSNV) == 0 && length(fNameCNA) == 0) {
+        stop("Input vectors are empty!")
+    }
+    cR <- character()
+    for (fSNV in fNameSNV) {
```

Comment:
Maybe just my taste but I don't **like** function behaviour that depends on file extensions. If a file is not RDS, then readRDS will throw an error ... I like to catch reasonable errors, but in this case - hm.